pytest/tests/loadtest/locust/common/ft.py

```diff
@@ -85,20 +86,31 @@ def create_passive_users(self,
         assert prefix_len > 4, f"user key {parent.key.account_id} is too long"
         chars = string.ascii_lowercase + string.digits
 
-        def create_account():
-            prefix = ''.join(random.choices(chars, k=prefix_len))
+        def create_account(i):
+            prefix = ''.join(random.Random(i).choices(chars, k=prefix_len))
             account_id = f"{prefix}.{parent.key.account_id}"
             return Account(key.Key.from_seed_testonly(account_id))
 
-        accounts = [create_account() for _ in range(num)]
-        node.prepare_accounts(accounts,
-                              parent,
-                              balance=1,
-                              msg="create passive user")
-        with futures.ThreadPoolExecutor() as executor:
-            futures.wait(
-                executor.submit(self.register_passive_user, node, account)
-                for account in accounts)
+        with futures.ThreadPoolExecutor(max_workers=4) as executor:
+            batch_size = 500
+            num_batches = (num + batch_size - 1) // batch_size
+            for i in range(num_batches):
+                accounts = [
+                    create_account(i)
+                    for i in range(i * batch_size, min((i + 1) *
+                                                       batch_size, num))
+                ]
+                node.prepare_accounts(accounts,
+                                      parent,
+                                      balance=1,
+                                      msg="create passive user")
+                futures.wait([
+                    executor.submit(self.register_passive_user, node, account)
+                    for account in accounts
+                ])
```

Comment:
```suggestion
                futures.wait(
                    executor.submit(self.register_passive_user, node, account)
                    for account in accounts
                )
```

No need to collect into a list before passing to `futures.wait`

Reply:
Thanks, done!